client/src/config/globalConfig.js

```diff
@@ -1,11 +1,11 @@
 
 const {
-  REACT_APP_SERVER_HOST: serverHost = 'http://localhost',
-  REACT_APP_SERVER_PORT: serverPort = 3001,
+  REACT_APP_SERVER_HOST: serverHost = 'http://localhost:3001',
+  REACT_APP_SERVER_PORT: serverPort,
 } = process.env;
 
 export default {
-  serverUrl: `${serverHost}:${serverPort}`,
+  serverUrl: serverPort ? `${serverHost}:${serverPort}` : `${serverHost}`,
```

Comment:
Better to get server URL directly, and default it to `http://localhost:3001` which is for local development.
There seems to be little use of separating the two at this point.